Dashboard/app/js/lib/views/devices/assignments-list-view.jsx

```diff
@@ -1,17 +1,113 @@
 import React from 'react';
-import dayjs from 'dayjs';
+
+import AssignmentsList from 'akvo-flow/components/devices/AssignmentsList';
+// import EditAssignment from 'akvo-flow/components/devices/EditAssignment';
+import observe from '../../mixins/observe';
 
 require('akvo-flow/views/react-component');
 
-FLOW.AssignmentsListView = FLOW.ReactComponentView.extend({
+FLOW.AssignmentsListView = FLOW.ReactComponentView.extend(observe({
+  'FLOW.surveyAssignmentControl.content.isLoaded': 'assignmentsList',
+  'FLOW.surveyAssignmentControl.content.length': 'assignmentsList',
+  'FLOW.dialogControl.delAssignmentConfirm': 'assignmentDelete',
+}), {
+  init() {
+    this._super();
+    this.assignmentsList = this.assignmentsList.bind(this);
+    this.assignmentEdit = this.assignmentEdit.bind(this);
+    this.assignmentDeleteConfirm = this.assignmentDeleteConfirm.bind(this);
+    this.assignmentSort = this.assignmentSort.bind(this);
+  },
+
   didInsertElement(...args) {
     this._super(...args);
+    const self = this;
```

Comment:
with arrow functions you dont need to do this anymore - see here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#No_separate_this